test/k8sT/istio.go

```diff
@@ -0,0 +1,325 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sTest
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/cilium/cilium/test/ginkgo-ext"
+	"github.com/cilium/cilium/test/helpers"
+
+	. "github.com/onsi/gomega"
+)
+
+// This tests the Istio 1.0.0 integration, following the configuration
+// instructions specified in the Istio Getting Started Guide in
+// Documentation/gettingstarted/istio.rst.
+// Changes to the Getting Started Guide may require re-generating or copying
+// the following manifests:
+// - istio-cilium.yaml
+// - bookinfo-v1-istio.yaml
+// - bookinfo-v2-istio.yaml
+// - istio-sidecar-init-policy.yaml
+// Cf. the comments below for each manifest.
+var _ = Describe("K8sIstioTest", func() {
+
+	var (
+		// istioSystemNamespace is the default namespace into which Istio is
+		// installed.
+		istioSystemNamespace = "istio-system"
+
+		// istioYAMLPath is the istio-cilium.yaml file generated following the
+		// instructions in Documentation/gettingstarted/istio.rst to setup
+		// Istio 1.0.0. mTLS is enabled.
+		istioYAMLPath = helpers.ManifestGet("istio-cilium.yaml")
+
+		// istioServiceNames is the subset of Istio services in the Istio
+		// namespace that are accessed from sidecar proxies.
+		istioServiceNames = []string{
+			// "istio-galley",
+			// "istio-egressgateway",
+			// "istio-ingressgateway",
+			// "istio-policy",
+			// "istio-telemetry",
+			"istio-statsd-prom-bridge",
+			"istio-pilot",
+			// "prometheus",
+			// "istio-citadel",
+			// "istio-sidecar-injector",
+		}
+
+		// wgetCommand is the command used in this test because the Istio apps
+		// do not provide curl.
+		wgetCommand = fmt.Sprintf("wget --tries=2 --connect-timeout %d", helpers.CurlConnectTimeout)
+
+		kubectl          *helpers.Kubectl
+		microscopeCancel func() error
```

Comment:
Initialize this to be the following:
```
microscopeCancel = func() error { return nil }
```

To ensure it is not nil in case callers try to invoke it before it is populated.

Reply:
Done.